plugins/stack-overflow/src/search/StackOverflowQuestionsCollator/StackOverflowQuestionsCollator.tsx

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { IndexableDocument, DocumentCollator } from '@backstage/search-common';
+import { Config } from '@backstage/config';
+import fetch from 'cross-fetch';
+import qs from 'qs';
+import {
+  StackOverflowQuestion,
+  StackOverflowQuestionsRequestParams,
+} from '../../types';
+
+interface StackOverflowDocument extends IndexableDocument {
+  answers: number;
+  tags: string[];
+}
+
+export class StackOverflowQuestionsCollator implements DocumentCollator {
+  protected baseUrl: string;
+  protected requestParams: StackOverflowQuestionsRequestParams;
+  public readonly type: string = 'stack-overflow';
+
+  constructor({
+    config,
+    requestParams,
+  }: {
+    config: Config;
+    requestParams: StackOverflowQuestionsRequestParams;
```

Comment:
Might i suggest making `fetchApi: FetchApi` injectable? And then using that to make requests. Even though the majority of its value comes out when talking to internal things, it can still be housing metrics, tracing, etc that an org wants to track. So it'd be nice to start using it in new things!

Reply:
Sure, sounds promising, will take a look!

Reply:
I have now resolved all your comments except this one. The reason why I did not change this is because 1. I couldn't find any documentation on the value of it, when browsing around the code I could see that auth is the main example. 2. I couldn't find any examples of backend plugins using it. Let me know if you still think it could be valuable to use it for the requests, and if so please point me to somewhere I can read up on it more. 🙏